src/components/tx/SignOrExecuteForm/SignerForm/index.tsx

```diff
@@ -0,0 +1,114 @@
+import {
+  Box,
+  FormControl,
+  InputLabel,
+  MenuItem,
+  Select,
+  SvgIcon,
+  Tooltip,
+  Typography,
+  type SelectChangeEvent,
+} from '@mui/material'
+import { useNestedSafeOwners } from '@/hooks/useNestedSafeOwners'
+import { useWalletContext } from '@/hooks/wallets/useWallet'
+import EthHashInfo from '@/components/common/EthHashInfo'
+import { useCallback, useContext, useMemo } from 'react'
+import useSafeInfo from '@/hooks/useSafeInfo'
+import TxCard from '@/components/tx-flow/common/TxCard'
+import InfoIcon from '@/public/images/notifications/info.svg'
+import SignatureIcon from '@/public/images/transactions/signature.svg'
+
+import css from './styles.module.css'
+import { sameAddress } from '@/utils/addresses'
+import { SafeTxContext } from '@/components/tx-flow/SafeTxProvider'
+import { MODALS_EVENTS, trackEvent } from '@/services/analytics'
+
+export const SignerForm = ({ willExecute }: { willExecute?: boolean }) => {
+  const { signer, setSignerAddress, connectedWallet: wallet } = useWalletContext() ?? {}
+  const nestedSafeOwners = useNestedSafeOwners()
+  const signerAddress = signer?.address
+  const { safe } = useSafeInfo()
+  const { safeTx } = useContext(SafeTxContext)
+
+  const onChange = (event: SelectChangeEvent<string>) => {
+    trackEvent(MODALS_EVENTS.CHANGE_SIGNER)
+    setSignerAddress?.(event.target.value)
+  }
+
+  const isNotNestedOwner = useMemo(() => nestedSafeOwners && nestedSafeOwners.length === 0, [nestedSafeOwners])
```

Comment:
Nit: There is a hook `useIsNestedSafeOwner` that can be used here.

Reply:
Addressed in 56de0ed1f7cc052f870738d6184a6b7e77981ca1.